### PR TITLE
beautiful table output included in the end

### PR DESCRIPTION
**Pull request by smpawar**

### CODE REVIEW

**Comment by smpawar** [comment]
@ersantosnet It is ready for merging to Main. Can you please review it

**Comment by smpawar** [comment]
@ersantosnet I have rebased  beautifultable_smpawar with latest main in this commit.
It is good for merging to main. I have tested it and works fine. Kindly review and merge to main

File: db_assessment/import_db_assessment.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
 import json
 import os
 import glob
+from sre_constants import SUCCESS
```
Comment (ersantosnet) [inline db_assessment/import_db_assessment.py:20]: Do we really need this one?

File: db_assessment/import_db_assessment.py [hunk 2]
```diff
@@ -518,20 +533,25 @@ def importCSVToBQ(gcpProjectName,bqDataset,tableName,fileName,skipLeadingRows,au
             print ('\n FAILED: Optimus Prime could not import the filename "{}" into "{}" because of the error "{}".\n'.format(fileName,table_id,importErr))
 
             print ('   Table Schema = {}'.format(schema))
-            return False
+            if 'csv' not in fileName :
```
Comment (ersantosnet) [inline db_assessment/import_db_assessment.py:536]: Do we really need this?

File: db_assessment/import_db_assessment.py [hunk 3]
```diff
@@ -553,6 +573,14 @@ def getObjNameFromFiles(fileName,splitterChar,pos):
     
     return None    
 
+def getPkeyFromFile(fileName):
```
Comment (ersantosnet) [inline db_assessment/import_db_assessment.py:576]: Consider using:

tableName = getObjNameFromFiles(fileName,'__',1)

File: db_assessment/import_db_assessment.py [hunk 4]
```diff
@@ -662,4 +690,103 @@ def insertErrors(invalidfiles,op_df,gcpProjectName,bq_dataset):
...
+    if btsource=='fromimportDataframeToBQ':
+
+        if dataframeornot is not None:
+            if 'PKEY' in df.columns.to_list():
+                # df2.reset_index(inplace=True)
+                if 'dbsizing_summary' not in tableName: # Need to add code to support dbsizing_summary or fix the issue with it 
+                    pkeygroupby=df.groupby(['PKEY']).size()
+                    # pkeygroupby=df2.groupby(level=0).size()
+                    pkeycount=pkeygroupby.to_dict()
+                    for pkeyname,pkeyname_rowcount in pkeycount.items():
```
Comment (ersantosnet) [inline db_assessment/import_db_assessment.py:708]: Can we work around the hardcoded stuff?

File: db_assessment/import_db_assessment.py [hunk 5]
```diff
@@ -662,4 +690,103 @@ def insertErrors(invalidfiles,op_df,gcpProjectName,bq_dataset):
         print ('\nWARNING: Issues while pusing Errors into operrors table with error ', pushErr)
 
 
+def populateBT(tableName,df,dataframeornot,invalidfiles,btsource,rowsimported,importresults):
+    # Fuction to populate the importresults list which will be used to print using Beautiful Table 
+    # rowsimported of <0 is used to indicate a FAILED status 
+    # fileName='dummy'
+    tmpdataFrame=pd.DataFrame()
+
+    if btsource=='fromimportDataframeToBQ':
```
Comment (ersantosnet) [inline db_assessment/import_db_assessment.py:699]: Can we use args instead?

File: db_assessment/import_db_assessment.py [hunk 6]
```diff
@@ -662,4 +690,103 @@ def insertErrors(invalidfiles,op_df,gcpProjectName,bq_dataset):
...
+                        if 'Elapsed' not in pkeyname:
+                            # twoDimensionalList.append([tableName,pkeyname,'SUCCESS',pkeyname_rowcount])    
+                            tmpdataFramedict = {"Target Table":tableName,"Distinct Pkey":pkeyname,"Import Status":"SUCCESS","Loaded rows":pkeyname_rowcount}
+                            tmpdataFrame = tmpdataFrame.append(tmpdataFramedict, ignore_index = True)  
+                            if len(tmpdataFrame) >0:
+                                importresults = pd.concat([importresults, tmpdataFrame], ignore_index = True, axis = 0)
+
+    elif btsource=='fromimportCSVToBQ':
+        fileName=invalidfiles
+        if 'opdbt' not in fileName: # to go past the error seen during Demo to Eri on Apr 26, 2022
```
Comment (ersantosnet) [inline db_assessment/import_db_assessment.py:718]: Remove comments

File: db_assessment/import_db_assessment.py [hunk 7]
```diff
@@ -662,4 +690,103 @@ def insertErrors(invalidfiles,op_df,gcpProjectName,bq_dataset):
...
+    #         importresults = pd.concat([importresults, tmpdataFrame], ignore_index = True, axis = 0)
+
+    return importresults
+
+def processBTResults(importresults):
+
+    #Create a in-memory database and connection
+    #Database is used for 
+    conn=sqlite3.connect(':memory:')
+    c=conn.cursor()
```
Comment (ersantosnet) [inline db_assessment/import_db_assessment.py:750]: Let's change it to pandas later

**Review by ersantosnet — APPROVED** [review]
Thank YOU!

File: db_assessment/import_db_assessment.py (hunk 1 above)
Comment (smpawar) [reply db_assessment/import_db_assessment.py]: This was fixed with the last commit

File: db_assessment/import_db_assessment.py (hunk 2 above)
Comment (smpawar) [reply db_assessment/import_db_assessment.py:535]: It is removed in latest commit for bug132 branch

File: db_assessment/import_db_assessment.py (hunk 3 above)
Comment (smpawar) [reply db_assessment/import_db_assessment.py:575]: getPkeyFromFile  is removed in latest commit for bug132 branch

File: db_assessment/import_db_assessment.py (hunk 5 above)
Comment (smpawar) [reply db_assessment/import_db_assessment.py:698]: args are used in latest commit for bug132 branch

File: db_assessment/import_db_assessment.py (hunk 6 above)
Comment (smpawar) [reply db_assessment/import_db_assessment.py:717]: comments are removed in latest commit for bug132 branch

File: db_assessment/import_db_assessment.py (hunk 7 above)
Comment (smpawar) [reply db_assessment/import_db_assessment.py:749]: changed to pandas in latest commit for bug132 branch

File: db_assessment/import_db_assessment.py (hunk 4 above)
Comment (ersantosnet) [reply db_assessment/import_db_assessment.py:707]: dbsizing_summary + Elapsed

Comment (smpawar) [reply db_assessment/import_db_assessment.py:707]: hardcoded 'dbsizing_summary' is removed in latest commit for bug132 branch

**Pull request merged**